go-controller/pkg/factory/factory.go

```diff
@@ -461,10 +463,71 @@ func NewWatchFactory(c kubernetes.Interface, stopChan chan struct{}) (*WatchFact
 		return nil, err
 	}
 
+	if err := wf.startup(stopChan); err != nil {
+		return nil, err
+	}
+	return wf, nil
+}
+
+// NewNodeWatchFactory creates only the informers we need for node-level operations,
+// saving bandwidth and memory
+// It only configures informers for:
+// - pods, and only the pods on this node
+// - services
+// - endpoints
+// - nodes
+func NewNodeWatchFactory(nodeName string, c kubernetes.Interface, stopChan chan struct{}) (*WatchFactory, error) {
+	wf := &WatchFactory{
+		iFactory:  informerfactory.NewSharedInformerFactory(c, resyncInterval),
+		informers: make(map[reflect.Type]*informer),
+	}
+	var err error
+
+	// We want to tweak pod lists, so we can't use the shared InformerFactory
+	podSharedInformer := v1coreinformers.NewFilteredPodInformer(
```

Comment:
The API docs say:
```
 NewFilteredPodInformer constructs a new informer for Pod type. Always prefer using an informer factory to get a shared informer instead of getting an independent one. This reduces memory footprint and number of connections to the server. 
```
So this is a new informer, not a shared index informer.
If this is intended, we should probably adjust this to `podInformer` vs. `podSharedInformer`.

Seeing as we have another connection to the API server, another cache etc... I'm not sure that this is much of an improvement over the current code :thinking:

Reply:
There is still a single informer and index for the whole process; since we mux our "informer" on top of the client-go informer, there's no actual duplication.

Nothing is shared between informers of different types, so that comment holds if we were creating lots of client-go pod informers. We're not, however, so this is fine.

It's impossible to create a SharedInformerFactory with a filter on one of the watches, BTW. I filed https://github.com/kubernetes/client-go/issues/801 to get around this.

As for connections, golang itself maintains a singleton of http/2 connections, so we don't actually have any duplicate sockets.